### PR TITLE
WIP dind: Fail if docker using non-shared mount propagation

### PR DESCRIPTION
**Pull request by marun**
The host docker daemon must run with shared mount propagation for
volume mounts to work correctly under dind.

cc: @openshift/networking

### CODE REVIEW

**Comment by marun** [comment]
[testextended][extended:networking]

**Comment by danwinship** [comment]
failure was spurious. [testextended][extended:networking]

File: hack/dind-cluster.sh [hunk 1]
```diff
@@ -138,6 +138,19 @@ function docker-exec-script() {
     ${DOCKER_CMD} exec -t "${cid}" ${BASH_CMD} ${cmd}
 }
 
+# Ensure docker is configured with shared mount propagation
+function check_docker_mount_flags() {
+  local flags="$(systemctl show docker.service | grep MountFlags |
+      sed -e 's+MountFlags=++')"
```
Comment (danwinship) [inline hack/dind-cluster.sh]: `"$(systemctl show docker.service | sed -ne 's+MountFlags=++p')"`

File: hack/dind-cluster.sh [hunk 2]
```diff
@@ -138,6 +138,19 @@ function docker-exec-script() {
...
+  # The default of shared will be used if the MountFlags is unset.
+  local unset="0"
+  local shared="1048576"
```
Comment (danwinship) [inline hack/dind-cluster.sh]: maybe `local shared=$((0x100000))` ?

Comment (danwinship) [inline hack/dind-cluster.sh]: or `$((1 << 20))`. and a note pointing out that it comes from sys/mount.h

**Comment by danwinship** [comment]
so, I have MountFlags=slave but dind seems to work fine for me... ?

**Comment by danwinship** [comment]
and now jenkins fails with:

> ERROR: The docker daemon is not running with MountFlags=shared.

**Comment by marun** [comment]
Tests were failing locally for me and clearing MountFlags got it working again, but if this breaks ci there must be something else involved.  Marking WIP for now.

**Comment by danwinship** [comment]
> Tests were failing locally for me and clearing MountFlags got it working again

maybe some other change also took effect when you restarted docker?

**Comment by marun** [comment]
Nobody seems to be having the problem this was supposed to fix, closing.

**Pull request closed without merging**